src/PluginStore.tsx

```diff
@@ -0,0 +1,23 @@
+import { IPlugin } from './interfaces/IPlugin';
+
+export class PluginStore {
+  private functionArray: Map<string, any>;
+
+  constructor() {
+    this.functionArray = new Map<string, any>();
+  }
+
+  install(plugin: IPlugin) {
+    plugin.init(this);
+    plugin.activate();
```

Comment:
Set the plugin to the map. Get the name from the plugin instance. The name of the plugin becomes the key of the map.